authors/apps/comments/views.py

```diff
@@ -0,0 +1,142 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from django.shortcuts import get_object_or_404
+
+from authors.apps.articles.models import Article
+from .models import Comment
+
+from .serializers import CommentSerializer, RecursiveField
+from .renderers import CommentJSONRenderer
+
+from django.core.serializers.json import DjangoJSONEncoder
+import json
+from drf_yasg.utils import swagger_auto_schema
+
+
+class CommentAPIView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (CommentJSONRenderer,)
+    serializer_class = CommentSerializer
+
+    @swagger_auto_schema(
+        operation_description='Create a Comment.',
+        operation_id='Comment on an article',
+        request_body=serializer_class,
+        responses={201: serializer_class(many=False), 400: 'BAD REQUEST'},
+    )
+    def post(self, request, *args, **kwargs):
+        comment = request.data.get('comment', {})
+        slug = kwargs.get('slug')
+        parent = request.data.get('comment', None).get('parent_id', None)
+        article = self.get_article(slug)
+        data = {**comment, "parent": parent}
+        context = {"author": request.user, "article": article}
+        serializer = self.serializer_class(data=data, context=context)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status.HTTP_201_CREATED)
+
+    def get(self, request, **kwargs):
+        slug = kwargs.get('slug')
+        article = self.get_article(slug)
+        queryset = Comment.objects.filter(article=article)
+        serializer = self.serializer_class(queryset, many=True)
+        return Response(serializer.data, status.HTTP_200_OK)
+    
+
+    def get_article(self, slug):
+        try:
+            article = get_object_or_404(Article, slug=slug)
+        except:
+            raise exceptions.ValidationError("This article does not exist!")
+        return article
+
+
+class ManageCommentAPIView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (CommentJSONRenderer,)
+    serializer_class = CommentSerializer
+
+    @swagger_auto_schema(
+        operation_description='Edit Comment.',
+        operation_id='Edit a comment on an article',
+        request_body=serializer_class,
+        responses={201: serializer_class(many=False), 400: 'BAD REQUEST'},
+    )
+    def patch(self, request, *args, **kwargs):
+        slug = kwargs.get('slug')
+        comment_id = kwargs.get('pk')
+        article = self.get_article(slug)
+        comment = Comment.objects.filter(article=article, id=comment_id).first()
+        if not comment:
+            return Response({
+                "error": "This article does not have this comment"
+            }, status.HTTP_404_NOT_FOUND)
+        current_user = request.user
+        comment = Comment.objects.filter(article=article, id=comment_id, author=current_user).first()
```

Comment:
This seems like repetition too, you already got the comment on line 73, comparing the author of the comment with the current user would be a better approach

Reply:
This is a sort of validation, at line 90 is to validate if the comment to edit exists, then here is to check if it was posted by the deletion requester.